mechanicalsoup/browser.py

```diff
@@ -139,8 +139,17 @@ def post(self, *args, **kwargs):
         Browser.add_soup(response, self.soup_config)
         return response
 
-    def _request(self, form, url=None, **kwargs):
-        """Extract input data from the form to pass to a Requests session."""
+    @staticmethod
+    def _get_request_kwargs(method, url, **kwargs):
+        """This method exists to raise a TypeError when a method or url is
+        specified in the kwargs.
+        """
+        request_kwargs = {"method": method, "url": url}
+        request_kwargs.update(kwargs)
+        return request_kwargs
+
+    @classmethod
+    def get_request_kwargs(cls, form, url=None, **kwargs):
```

Comment:
Since this is a public method, it will need a docstring.

Reply:
Hmm, and this should be a @staticmethod, not @classmethod, right?

Reply:
I think they used classmethod because it calls
```python
return cls._get_request_kwargs(method, url, files=files, **kwargs)
```
However, this could be `Browser._get_request_kwargs`. I'm happy with it changed or as-is.

Reply:
OK, I'm fine with both too.

Reply:
I added a simple docstring to the public method.